xlml/apis/task.py

```diff
@@ -70,14 +70,39 @@ def run(self) -> DAGNode:
         group_id=self.task_test_config.benchmark_id, prefix_group_id=True
     ) as group:
       provision, queued_resource, ssh_keys = self.provision()
-      run_model = self.run_model(queued_resource, ssh_keys)
       post_process = self.post_process()
       clean_up = self.clean_up(queued_resource)
-
+      run_model = self.run_model(queued_resource, ssh_keys)
       provision >> run_model >> post_process >> clean_up
 
     return group
 
+  def run_with_startup_script(self) -> DAGNode:
+    """Run a test job on GCE with startup script.
+
+    Returns:
+      A task group with the following tasks chained:
+      provision_with_startup_script (create_queued_resource_request + wait_for_ready_queued_resource + check_if_startup_script_end),
+      post_process and clean_up.
+    """
+
+    use_startup_script = True
+
+    with TaskGroup(
+        group_id=self.task_test_config.benchmark_id, prefix_group_id=True
+    ) as group:
+      (
+          provision_with_startup_script,
+          queued_resource,
+          ssh_keys,
+      ) = self.provision_with_startup_script(use_startup_script=use_startup_script)
```

Comment:
We don't need `use_startup_script` param here as it will always be True.

Reply:
Good catch! Done.